### DISCUSSION TITLE
How to forecast for future time period

### DISCUSSION THREAD

**Discussion by nitesh31mishra** [awslabs/gluonts#1550] [Q&A]
Actually, I have data for the year 2019-20 (two years) and I have made a model on this particular data. Now I want my model to forecast the new time period from Jan-2021 to march-2021. I have gone through documentation and GitHub but couldn't find anything for these. How can I do it ??? 

about data :
My data contain multivariate time series and we are trying to forecast it ...
I have 5 feat_static_cat and 10 feat_dynamic_real...

@mbohlkeschneider 
@lostella @timoschowski

**Comment by mbohlkeschneider** [comment]
Hi @nitesh31mishra,

Thank you for your question. Can you provide minimal reproducible example that illustrates what the issue is and what you intend to do? 

In a nutshell, it should be something like this: 


```
from gluonts.model.deepar import DeepAREstimator
from gluonts.mx.trainer import Trainer
from gluonts.dataset.repository.datasets import get_dataset

ds = get_dataset('m4_hourly')
estimator = DeepAREstimator(freq=ds.metadata.freq, prediction_length=24, trainer=Trainer(epochs=1))
predictor = estimator.train(ds.train)
forecast_it = predictor.predict(ds.train)

for forecast in forecast_it:
    pass 
    # do something with the forecast object
```

Thank you!

**Reply by MAYUKH97** [reply]
Hey @mbohlkeschneider actually we have a data set where we have like item_id for different types of Food around 500 and data set has around (100000,35) rows and columns,and we have categorical as well as continuous values which we have input them as (feat_dynamic_real, for continous and feat_static_cat for the categorical columns) then we made 500 different time series for each item_id and we have done evaluation and got the metrics for RMSE,MSE,etc now as @nitesh31mishra said we have data from jan,2019-dec,2020 and now we want to forecast for the year from january 2021- march,2021 which has not occurred and want to forecast what will be the Revenue for next 1 year. so can you help us with the logic and code for this.

**Reply by pratikgehlott** [reply]
hey, @mbohlkeschneider @lostella any suggestion?? even I have the same issue.
